### PR TITLE
Remove EnkfNode and related C++ code

### LINKED ISSUE #4305: Remove enkf_node.cpp
After all enkf_node types has been removed, we can clean up some unused C++ code

### PR DESCRIPTION
**Pull request by ManInFez**
**Issue**
Resolves #4305 

**Approach**
_Short description of the approach_

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: tests/unit_tests/c_wrappers/res/enkf/test_field_export.py [hunk 1]
```diff
@@ -32,33 +30,37 @@ def test_field_basics(snake_oil_field_example):
...
 
 
 @pytest.mark.skip(reason="pending refactor")
 def test_field_export_many(snake_oil_field_example):
-    ert = snake_oil_field_example
-    fs_manager = ert.storage_manager
-    ens_config = ert.ensembleConfig()
-    config_node = ens_config["PERMX"]
+    pass
+    # ert = snake_oil_field_example
```
Comment (oyvindeide) [inline tests/unit_tests/c_wrappers/res/enkf/test_field_export.py:50]: Some commented out code here

Comment (ManInFez) [reply tests/unit_tests/c_wrappers/res/enkf/test_field_export.py:50]: yes just left it there for documenting previous behavior. Can remove the comments :shrug:

File: src/ert/gui/tools/manage_cases/case_init_configuration.py [hunk 2]
```diff
@@ -175,9 +174,7 @@ def initializeFromExisting(_):
                     member_mask[int(member)] = True
                 source_fs = case_manager[source_case_name]
                 target_fs = case_manager[target_case_name]
-                source_fs.copy_from_case(
-                    target_fs, report_step, parameters, member_mask
```
Comment (oyvindeide) [inline src/ert/gui/tools/manage_cases/case_init_configuration.py:179]: Could you add in the commit body why this has been changed?

**Review by oyvindeide — APPROVED** [review]
Just a minor comment, otherwise, good job 👍

**Review by pinkwah — APPROVED** [review]
Looks really cool!!!! But comment 😔

File: tests/unit_tests/c_wrappers/res/enkf/test_field_export.py [hunk 3]
```diff
@@ -36,8 +34,8 @@ def test_field_export(snake_oil_field_example):
     fs_manager = ert.storage_manager
```
[the remark this answers is not in the document]
Comment (ManInFez) [reply tests/unit_tests/c_wrappers/res/enkf/test_field_export.py:34]: which comments?

**Pull request merged**